src/App.tsx

```diff
@@ -1,24 +1,208 @@
 /* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  USER_ID, addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { List } from './components/List/List';
+import { Footer } from './components/Footer/Footer';
+import { handleError } from './handleError';
+import { ErrorMessage } from './types/ErrorMessage';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const [filter, setFilter] = useState<Filter>('All');
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+  const [isToggling, setIsToggling] = useState<boolean>(false);
+  const [isTogglingAll, setIsTogglingAll] = useState<boolean>(false);
+
+  const counter = () => {
+    return todos.filter(todo => !todo.completed).length;
+  };
+
+  const visibleTodos = () => {
+    const visible = [...todos].filter(todo => {
+      if (filter === 'Active' && todo.completed) {
+        return false;
+      }
+
+      if (filter === 'Completed' && !todo.completed) {
+        return false;
+      }
+
+      return true;
+    });
+
+    return visible;
+  };
+
+  const handleSubmit = (event: { preventDefault: () => void }) => {
```

Comment:
try to use Event type: https://bobbyhadz.com/blog/typescript-react-onsubmit-event-type